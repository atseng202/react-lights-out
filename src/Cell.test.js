import React from "react";

const flipCellsAroundMe = jest.fn();

import { render } from "@testing-library/react";
import Cell from "./Cell";

/* ******************** Smoke Tests  ******************************* */
describe("smoke tests for Cell", function () {
  it("renders without crashing", function () {

    render(
      <table>
        <tbody>
          <tr>
            <Cell flipCellsAroundMe={flipCellsAroundMe} isLit={true} />
          </tr>
        </tbody>
      </table>
    );
  });
  // end
});

/* ******************** Snapshot Tests  ******************************* */
// NOTE: is there a better way to test that Cell renders correctly
// <td> cannot be rendered in isolation
describe("snapshot for individual cell", function () {
  it("matches a lit Cell", function () {

    const { container, debug } = render(
      <table>
        <tbody>
          <tr>
            <Cell flipCellsAroundMe={flipCellsAroundMe} isLit={true} />
          </tr>
        </tbody>
      </table>
    );
    expect(container
      .querySelector("tr")
      .firstChild
      .classList
      .contains("Cell-lit")).toEqual(true);

    expect(container).toMatchSnapshot();
  });

  it("matches an unlit Cell", function () {
    const { container, debug } = render(
      <table>
        <tbody>
          <tr>
            <Cell flipCellsAroundMe={flipCellsAroundMe} isLit={false} />
          </tr>
        </tbody>
      </table>
    );
    expect(container
      .querySelector("tr")
      .firstChild
      .classList
      .contains("Cell-lit")).toEqual(false);
    expect(container).toMatchSnapshot();
  });
  // end
});

