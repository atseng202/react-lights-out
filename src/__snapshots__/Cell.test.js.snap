// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for individual cell matches a lit Cell 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class="Cell Cell-lit"
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`snapshot for individual cell matches an unlit Cell 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class="Cell "
        />
      </tr>
    </tbody>
  </table>
</div>
`;
